python/mypy_protobuf.py

```diff
@@ -528,6 +529,77 @@ def write_services(self, services):
                 )
                 self.write_methods(service, is_abstract=False)
 
+    def _output_type(self, method):
+        # type: (d.MethodDescriptorProto) -> Text
+        result = self._import_message(method.output_type)
+        if method.server_streaming:
+            result = "{}[{}, None, None]".format(
+                self._import("typing", "Generator"), result,
+            )
+        return result
+
+    def write_grpc_methods(self, service):
+        # type: (d.ServiceDescriptorProto) -> None
+        l = self._write_line
+        methods = [m for m in service.method if m.name not in PYTHON_RESERVED]
+        if not methods:
+            l("pass")
+            l("")
+        for method in methods:
+            l("@{}", self._import("abc", "abstractmethod"))
+            l("def {}(self,", method.name)
+            with self._indent():
+                l("request: {},", self._import_message(method.input_type))
+                l("context: {},", self._import("grpc", "ServicerContext"))
+            l(
+                ") -> {}: ...", self._output_type(method),
+            )
+            l("")
+
+    def write_grpc_stub_methods(self, service):
+        # type: (d.ServiceDescriptorProto) -> None
+        l = self._write_line
+        methods = [m for m in service.method if m.name not in PYTHON_RESERVED]
+        if not methods:
+            l("pass")
+            l("")
+        for method in methods:
+            l("def {}(self,", method.name)
+            with self._indent():
+                l("request: {},", self._import_message(method.input_type))
+            l(
+                ") -> {}: ...", self._output_type(method),
+            )
+            l("")
+
+    def write_grpc_services(self, services):
+        # type: (Iterable[d.ServiceDescriptorProto]) -> None
+        l = self._write_line
+        l(
+            "from .{} import *",
```

Comment:
here I'm making an assumption that `mypy_out` and `mypy_grpc_out` are the same. We need somehow to use messages inside of services. How to do this better?

Reply:
Good callout - I'm not sure there's much we can do. You can see in our protoc instantiation in `run_test.sh` that there's already an assumption that py and mypy go to the same location. I think it's ok to make this assumption for now (no need to block on solving this)

Open to any creative ideas to enforce this.

Reply:
Ideally we wouldn't need this line - but rather above `self._import_message` would bring in mypy_out